src/Illuminate/Database/Connection.php

```diff
@@ -1265,6 +1275,8 @@ public function getReadPdo()
             return $this->getPdo();
         }
 
+        $this->retrievingPdoType('read');
```

Comment:
We could call this at the beginning of the method. If `$this->getPdo()` is called from this method, it would overwrite the `latestPdoTypeUsed`. Also, when `$this->readPdo` is null (i.e. read/write connections are not configured), it will call `$this->getPdo()` anyway.